### PR TITLE
Autosegmentation

### PR DESCRIPTION
**Pull request by lalmei**
## Description

Autosegmentation

### General Checklist

* [x] Tests added for this feature/bug
      if it was a bug, test must cover it.
* [x] Conform by the style guides, by using formatter
* [x] Documentation updated
* [x] (optional) Please add a label to your PR

### CODE REVIEW

File: src/whylogs/features/autosegmentation.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+
+def _entropy(series: pd.Series, normalized: bool = True):
```
Comment (andyndang) [inline src/whylogs/features/autosegmentation.py]: Can we have some unit test for these methods?

Comment (lalmei) [reply src/whylogs/features/autosegmentation.py]: will do

File: tests/unit/features/test_autosegmentation.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+import pytest
+
+from whylogs.features.autosegmentation import _entropy, _weighted_entropy,_estimate_segments
+
+# import numpy as np
+
+
+def test_entropy_ints():
+    series = pd.Series([0, 1, 2, 4])
```
Comment (andyndang) [inline tests/unit/features/test_autosegmentation.py:14]: can you add tests for empty series as well? Just make sure we handle edge cases

**Review by andyndang — APPROVED** [review]
LGTM
